src/lib/authui/sssd/main_dialog.rb

```diff
@@ -0,0 +1,517 @@
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact SUSE Linux GmbH.
+#
+# ------------------------------------------------------------------------------
+
+require "yast"
+require "auth/authconf.rb"
+require "authui/sssd/uidata.rb"
+require "authui/sssd/new_section_dialog.rb"
+require "authui/sssd/manage_ad_dialog.rb"
+require "authui/sssd/edit_param_dialog.rb"
+
+module SSSD
+    # Main dialog is split into overview on left side, and config editor on right side.
+    class MainDialog
+        include Yast
+        include Auth
+        include UIShortcuts
+        include I18n
+        include Logger
+
+        def initialize
+            textdomain "auth-client"
+            # SSSD section name to UI caption mapping
+            @sect_name_caption = {
+                "sssd" => _("Global option"), "nss" => _("Name switch"),
+                "pam" => _("Authentication"), "sudo" => _("Sudo"),
+                "autofs" => _("Auto-mount"), "ssh" => _("SSH public keys"),
+                "pac" => _("Privilege account certificate extension (MS-PAC)"),
+            }
+            # The reverse caption mapping
+            @sect_caption_name = Hash[*@sect_name_caption.map { |name, caption| [caption, name] }.flatten]
+        end
+
+        def run
+            return if !render_all
+            begin
+                return ui_event_loop
+            ensure
+                UI.CloseDialog()
+            end
+        end
+
+        private
+            # Render overview and edit buttons on left side, config editor on right side.
+            def render_all
+                UI.OpenDialog(
+                    Opt(:decorated, :defaultsize),
+                    VBox(
+                        Left(HBox(
+                            Icon::Simple("yast-nis"),
+                            Heading(_("Manage authentication domains via SSSD"))
+                        )),
+                        HBox(
+                            # Overview of all config sections
+                            HWeight(50, VBox(
+                                Frame(
+                                    _(""),
+                                    VBox(
+                                        Left(HBox(Label(_("Daemon status: ")), Label(Service.Active("sssd") ? _("Running") : _("Stopped")))),
+                                        Left(CheckBox(Id(:enable_daemon_pam), Opt(:notify), _("Allow domain users to authenticate on this computer"), AuthConfInst.sssd_enabled && AuthConfInst.sssd_pam)),
+                                         Left(HBox(
+                                            CheckBox(Id(:mkhomedir_enable), _('Automatically create home directory'), AuthConfInst.mkhomedir_pam),
+                                            Label(_('Current status:')),
+                                            Label(Id(:mkhomedir_status), AuthConfInst.mkhomedir_pam ? _('Enabled') : _('Disabled')),
```

Comment:
so this displays (BTW with no separating space)
```console
[ ] Automatically create home directoryCurrent status:Disabled
```
Why is a separate status label needed? I would expect that this is communicated already by the value of the checkbox.